src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/PinBasedOperationService.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.cmc.claimstore.events.claim;
+
+import com.google.common.collect.ImmutableList;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationsProperties;
+import uk.gov.hmcts.cmc.claimstore.documents.PrintService;
+import uk.gov.hmcts.cmc.claimstore.events.DocumentUploadHandler;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.ClaimIssuedNotificationService;
+import uk.gov.hmcts.cmc.claimstore.services.staff.ClaimIssuedStaffNotificationService;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import static java.util.Collections.singletonList;
+
+@Service
+@ConditionalOnProperty(prefix = "feature_toggles", name = "async_event_operations_enabled")
+public class PinBasedOperationService {
```

Comment:
This can be renamed better, as this might suggest that we have some non pin based operations to choose instead.

Reply:
renamed to `PinOrchastrationService`